### PR TITLE
[WIP] LAM-49 Init scripts.

### PR DESCRIPTION
**Pull request by gouzouni625**
This pull request contains the init scripts for Apache Zookeeper and Apache Kafka.

### CODE REVIEW

**Comment by gouzouni625** [comment]
This pull request will also include the commits of LAM-52.

File: ansible/roles/apache-hadoop/templates/hdfs-init.j2 [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    stop
+    ;;
+  restart|reload|force-reload)
+    restart
+    ;;
+  condrestart)
+    [ -f $LOCKFILE ] && restart || :
+    ;;
+  status)
+    ps -p $(cat $PIDFILE)
```
Comment (pkoro) [inline ansible/roles/apache-hadoop/templates/hdfs-init.j2]: `$PIDFILE` is not initialized somewhere above.

File: ansible/roles/apache-hadoop/templates/yarn-init.j2 [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    stop
+    ;;
+  restart|reload|force-reload)
+    restart
+    ;;
+  condrestart)
+    [ -f $LOCKFILE ] && restart || :
+    ;;
+  status)
+    ps -p $(cat $PIDFILE)
```
Comment (pkoro) [inline ansible/roles/apache-hadoop/templates/yarn-init.j2]: Again, `$PIDFILE` is not properly initialized.

**Comment by gouzouni625** [comment]
please review @pkoro .

**Comment by pkoro** [comment]
+1 . It seems excellent overall! My only comment is I don't see a `status` switch within yarn-init script. Should we have one?

File: ansible/roles/apache-flink/templates/flink-init.j2 [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+# The full path of the lock file to use.
+LOCKFILE="$INSTALLATION_PATH/flink/flink-lock"
+
+# The command that will start Apache Flink.
+START_COMMAND="$INSTALLATION_PATH/flink/bin/yarn-session.sh -n {{ number_of_taskmanagers }} -tm {{ ram_per_task_manager }}"
+
+# The command that will stop Apache Flink. Note that, HADOOP_HOME home variable should be set outside from this script and
+# before Flink's deployment.
+STOP_COMMAND="$HADOOP_HOME/bin/yarn application --kill"
```
Comment (skanct) [inline ansible/roles/apache-flink/templates/flink-init.j2:17]: Where is HADOOP_HOME set?

**Comment by gouzouni625** [comment]
I deliberately removed status switch within yarn-init script. To get status information about a yarn cluster, one needs to enable yarn high availability mode[1] which would introduce some changes in the whole cluster setup. I thought it would be better to make these changes only when we find out we really need to be able to get the status of the yarn cluster through the init file.

[1] https://hadoop.apache.org/docs/r2.4.1/hadoop-yarn/hadoop-yarn-site/YarnCommands.html#rmadmin

**Comment by pkoro** [comment]
Ok great! Thank for the clarification.

File: ansible/roles/apache-flink/templates/flink-init.j2 (hunk 3 above)
Comment (gouzouni625) [inline ansible/roles/apache-flink/templates/flink-init.j2:17]: HADOOP_HOME is an environment variable and a prerequisite for Apache Flink installation. It should be set up before Apache Flink is installed. At the current moment, it is defined within /etc/environment file by the Ansible common role[1].

https://github.com/grnet/okeanos-LoD/blob/devel/ansible/roles/common/files/environment

Comment (skanct) [inline ansible/roles/apache-flink/templates/flink-init.j2:17]: Is /etc/environment sourced during the boot process? You should either explicitly source /etc/environment or set the variable from ansible

**Comment by skanct** [comment]
Status should be supported somehow in my opinion

File: ansible/roles/apache-flink/templates/flink-init.j2 (hunk 3 above)
Comment (gouzouni625) [inline ansible/roles/apache-flink/templates/flink-init.j2:17]: /etc/environment is sourced by default during the user login so the variables declared in it, will be available. The same is done with JAVA_HOME variable which is used by Apache Hadoop.

Comment (skanct) [inline ansible/roles/apache-flink/templates/flink-init.j2:17]: Yes but my question was about the boot process, not about the user login. What happens when the VM is rebooted? Furthermore the init scripts should not rely on the environment variables of the the user running the scripts. This why I am asking you to either source the /etc/environment or declare the variables properly

Comment (gouzouni625) [inline ansible/roles/apache-flink/templates/flink-init.j2:17]: I have already changed the init scripts to source /etc/environment upon start.

**Pull request closed without merging**